tests/tests_geomstats/test_functions.py

```diff
@@ -0,0 +1,46 @@
+"""Unit tests for the functions manifolds."""
+
+import numpy as np
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.functions import SinfSpace
+
+
+class TestFunctionsSinf(geomstats.tests.TestCase):
+    def setup_method(self):
+        self.domain = gs.linspace(-gs.pi, gs.pi, num=50)
+        self.funcs = lambda a: np.sin(a * self.domain).reshape(1, 50)
+        self.f = self.funcs(1)
+        self.manifold = SinfSpace(self.domain)
+
+    @geomstats.tests.np_and_autograd_only
+    def test_manifold(self):
+        proj_f = self.manifold.projection(self.f)
+        result = self.manifold.belongs(proj_f)
+        self.assertTrue(result, "Expected True but got %s" % result)
+
+    @geomstats.tests.np_and_autograd_only
+    def test_metric(self):
```

Comment:
Could you add lower level tests (test_inner_product, test_exp, test_log, test_log_and_exp, etc?)

Reply:
Done!